compiler/luci-interpreter/pal/cmsisnn/PALAveragePool2d.h

```diff
@@ -94,19 +94,29 @@ inline void AveragePool<int8_t>(const tflite::PoolParams &params,
 }
 
 static inline void SetupScratchpadTensor(luci_interpreter::Tensor *scratchpad,
+                                         const luci_interpreter::DataType &input_data_type,
                                          const tflite::RuntimeShape &input_shape,
                                          const tflite::RuntimeShape &output_shape)
 
 {
-  assert(input_shape.DimensionsCount() == 4);
-  assert(output_shape.DimensionsCount() == 4);
+  if (input_data_type == luci_interpreter::DataType::S8)
+  {
+    assert(input_shape.DimensionsCount() == 4);
+    assert(output_shape.DimensionsCount() == 4);
+
+    const int32_t output_width = output_shape.Dims(2);
+    const int32_t depth = tflite::MatchingDim(input_shape, 3, output_shape, 3);
 
-  const int32_t output_width = output_shape.Dims(2);
-  const int32_t depth = tflite::MatchingDim(input_shape, 3, output_shape, 3);
+    const int32_t buf_size = arm_avgpool_s8_get_buffer_size(output_width, depth);
+    auto data_type_size = static_cast<int32_t>(luci_interpreter::getDataTypeSize(input_data_type));
 
-  const int32_t buf_size = arm_avgpool_s8_get_buffer_size(output_width, depth);
-  luci_interpreter::Shape scratchpad_shape{buf_size};
-  scratchpad->resize(scratchpad_shape);
+    luci_interpreter::Shape scratchpad_shape{buf_size * data_type_size};
```

Comment:
Note
From documentation `arm_avgpool_s8_get_buffer_size`: "The function returns required buffer size in bytes"

So multiplying by data_type_size is unnecessary